File: test-projects/java-sample/src/main/java/com/codegraph/sample/DatabaseConnection.java
package com.codegraph.sample;

import java.util.ArrayList;
import java.util.List;

/**
 * Database connection manager
 */
public class DatabaseConnection {
    private final String url;
    private boolean connected = false;

    /**
     * Create a new database connection
     *
     * @param url Database connection URL
     */
    public DatabaseConnection(String url) {
        this.url = url;
    }

    /**
     * Connect to the database
     */
    public void connect() {
        System.out.println("Connecting to database: " + url);
        try {
            Thread.sleep(100);
            connected = true;
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
    }

    /**
     * Disconnect from the database
     */
    public void disconnect() {
        System.out.println("Disconnecting from database");
        try {
            Thread.sleep(50);
            connected = false;
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
    }

    /**
     * Execute a query that returns results
     *
     * @param sql SQL query
     * @param params Query parameters
     * @return Query results
     * @throws IllegalStateException If database is not connected
     */
    public List<User> query(String sql, Object... params) {
        ensureConnected();
        System.out.println("Executing query: " + sql);
        try {
            Thread.sleep(10);
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
        return new ArrayList<>();
    }

    /**
     * Execute a query that doesn't return results
     *
     * @param sql SQL query
     * @param params Query parameters
     * @throws IllegalStateException If database is not connected
     */
    public void execute(String sql, Object... params) {
        ensureConnected();
        System.out.println("Executing statement: " + sql);
        try {
            Thread.sleep(10);
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
    }

    /**
     * Check if connected to database
     *
     * @return True if connected, false otherwise
     */
    public boolean isConnected() {
        return connected;
    }

    /**
     * Ensure database is connected
     *
     * @throws IllegalStateException If database is not connected
     */
    private void ensureConnected() {
        if (!connected) {
            throw new IllegalStateException("Database not connected");
        }
    }
}

File: test-projects/php-sample/src/DatabaseConnection.php
<?php

namespace CodeGraph\Sample;

/**
 * Database connection manager
 */
class DatabaseConnection
{
    private string $url;
    private bool $connected = false;

    /**
     * Create a new database connection
     *
     * @param string $url Database connection URL
     */
    public function __construct(string $url)
    {
        $this->url = $url;
    }

    /**
     * Connect to the database
     */
    public function connect(): void
    {
        echo "Connecting to database: {$this->url}\n";
        // Simulate async connection
        usleep(100000);
        $this->connected = true;
    }

    /**
     * Disconnect from the database
     */
    public function disconnect(): void
    {
        echo "Disconnecting from database\n";
        usleep(50000);
        $this->connected = false;
    }

    /**
     * Execute a query that returns results
     *
     * @param string $sql SQL query
     * @param array $params Query parameters
     * @return array Query results
     * @throws \RuntimeException If database is not connected
     */
    public function query(string $sql, array $params = []): array
    {
        $this->ensureConnected();
        echo "Executing query: {$sql}\n";
        usleep(10000);
        return ['rows' => [], 'rowCount' => 0];
    }

    /**
     * Execute a query that doesn't return results
     *
     * @param string $sql SQL query
     * @param array $params Query parameters
     * @throws \RuntimeException If database is not connected
     */
    public function execute(string $sql, array $params = []): void
    {
        $this->ensureConnected();
        echo "Executing statement: {$sql}\n";
        usleep(10000);
    }

    /**
     * Check if connected to database
     *
     * @return bool True if connected, false otherwise
     */
    public function isConnected(): bool
    {
        return $this->connected;
    }

    /**
     * Ensure database is connected
     *
     * @throws \RuntimeException If database is not connected
     */
    private function ensureConnected(): void
    {
        if (!$this->connected) {
            throw new \RuntimeException('Database not connected');
        }
    }
}
